cdl_utils/capdl_linker.py

```diff
@@ -63,11 +63,10 @@ def final_spec(args, obj_space, cspaces, addr_spaces, targets, architecture):
                     # We exclude TCBs that refer to a different CSpace
                     continue
                 funcs = {"get_vaddr": lambda x: elf.get_symbol_vaddr(x)}
-                s = EvalWithCompoundTypes(functions=funcs)
-                tcb.ip = s.eval(str(tcb.ip))
-                tcb.sp = s.eval(str(tcb.sp))
-                tcb.addr = s.eval(str(tcb.addr))
-                tcb.init = s.eval(str(tcb.init))
+                tcb.ip = eval(str(tcb.ip), {"__builtins__": None}, funcs)
+                tcb.sp = eval(str(tcb.sp), {"__builtins__": None}, funcs)
+                tcb.addr = eval(str(tcb.addr), {"__builtins__": None}, funcs)
+                tcb.init = eval(re.match("\[[0-9]*\]", str(tcb.init)).string)
```

Comment:
There are examples of tcb.init needing to call get_vaddr: https://github.com/seL4/sel4-tutorials/blob/master/tools/context.py#L143, but it doesn't look like funcs is passed into this eval call.

Reply:
Thanks for pointing out, fixed.